src/bentoml/_internal/frameworks/transformers.py

```diff
@@ -605,6 +643,337 @@ def make_default_signatures(pretrained: t.Any) -> ModelSignaturesType:
+
+    .. note::
+
+        Both arguments ``task_name`` and ``task_definition`` must be provided to set save a custom pipeline.
+
+    Returns:
+        A :obj:`~bentoml.Model` instance referencing a saved model in the local BentoML
+        model store.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+
+        bentoml.transformers.import_model(
+            'my_t5_model',
+            "t5-base",
+            signatures={
+                "__call__": {"batchable": False},
+            }
+        )
+    """
+
+    tag = Tag.from_taglike(name)
+
+    if sync_with_hub_version:
+        if tag.version is not None:
+            logger.warn(
+                f"sync_with_hub_version is True, user provided version {tag.version} may be overridden by huggingface hub's commit hash"
+            )
+
+    # The below API are introduced since 4.18
+    if pkg_version_info("transformers")[:2] >= (4, 18):
+        from transformers.utils import is_flax_available
+        from transformers.utils import is_tf_available
+        from transformers.utils import is_torch_available
+    else:
+        from .utils.transformers import is_flax_available
+        from .utils.transformers import is_tf_available
+        from .utils.transformers import is_torch_available
+
+    framework_versions = {"transformers": get_pkg_version("transformers")}
+    if is_torch_available():
+        framework_versions["torch"] = get_pkg_version("torch")
+    if is_tf_available():
+        from .utils.tensorflow import get_tf_version
+
+        framework_versions[
+            "tensorflow-macos" if platform.system() == "Darwin" else "tensorflow"
+        ] = get_tf_version()
+    if is_flax_available():
+        framework_versions.update(
+            {
+                "flax": get_pkg_version("flax"),
+                "jax": get_pkg_version("jax"),
+                "jaxlib": get_pkg_version("jaxlib"),
+            }
+        )
+    context = ModelContext(
+        framework_name="transformers", framework_versions=framework_versions
+    )
+
+    if signatures is None:
+        signatures = {
+            "__call__": {"batchable": False},
+        }
+        logger.info(
+            'Using the default model signature for Transformers (%s) for model "%s".',
+            signatures,
+            name,
+        )
+
+    config = transformers.AutoConfig.from_pretrained(
+        model_name_or_path,
+        proxies=proxies,
+        revision=revision,
+        variant=variant,
+        **kwargs,
+    )
+    model = None
+    downloaded_model_successfully = False
+    if os.path.isdir(model_name_or_path):
+        src_dir = model_name_or_path
+        if sync_with_hub_version:
+            raise BentoMLException(
+                "Cannot sync version with huggingface hub when importing a local model"
+            )
+    else:
+        try:
+            _try_import_accelerate()
+            from accelerate import init_empty_weights
```

Comment:
hmm this logics depends on accelerate is kinda weird.

Since it will probably end up going for snapshot_download nonetheless, since we can assume most people won't have accelerate install?

Reply:
@aarnphm can we just ask user to install accelerate if it's not available? I think accelerate is helpful for loading LLM models?

Reply:
afaik accelerate is mostly well known for ppl who train model with transformers. usually people won't like installing additional dependencies unless they have to.

I mean we can recommend them, sure, but we can't always rely that accelerate is always available unless it becomes `bentoml[transformers]` or `bentoml[huggingface]`

Reply:
@MingLiangDai how much work needed to fallback to your original implementation if accelerate is not available?

Reply:
I did a bit of digging into `init_empty_weights`. It seems I can easily extract that code out of the library for our use case (without any additional dependencies that we don't already have). So I will put `init_empty_weights` into our code base in `utils/transformers.py` and reference that instead.